File: api/src/main/java/com/api/service/WarehouseService.java
package com.api.service;

import com.api.dto.WarehouseDTO;
import com.api.dto.WarehouseIndexDTO;
import com.api.model.Warehouse;

import java.util.List;

public interface WarehouseService {

    Warehouse dtoToEntity(WarehouseDTO warehouseDTO);

    List<WarehouseDTO> findAll();

    Warehouse get(Long id);

    Warehouse save(WarehouseDTO warehouse);

    void delete(Long id);

    List<WarehouseDTO> index(WarehouseIndexDTO warehouseIndexDTO);
}

File: front-end/src/app/wr/warehouse/warehouse.service.js
(function() {

    angular.module('app.wr.warehouse').service('warehouseService', warehouseService);

    function warehouseService(warehouseFactory) {
        return {
            getWarehouses : getWarehouses,
            getWarehouse : getWarehouse,
            saveWarehouse : saveWarehouse,
            deleteWarehouse : deleteWarehouse,
            indexWarehouse : indexWarehouse
        }
        
        function getWarehouses() {
            return warehouseFactory.list({});
        }

        function getWarehouse(id) {
            return warehouseFactory.get({id},null);
        }

        function saveWarehouse(warehouse) {
            return warehouseFactory.save({}, warehouse);
        }

        function deleteWarehouse(id) {
            return warehouseFactory.delete({id},null);
        }

        function indexWarehouse(warehouseArgs) {
            return warehouseFactory.index({},warehouseArgs);
        }
    }

})();